dex/src/main/scala/com/wavesplatform/dex/api/websockets/WsClientMessage.scala

```diff
@@ -0,0 +1,25 @@
+package com.wavesplatform.dex.api.websockets
+
+import play.api.libs.json._
+
+trait WsClientMessage extends WsMessage
+
+object WsClientMessage {
+  implicit val wsClientMessageReads: Reads[WsClientMessage] = Reads { json =>
+    (json \ "T").asOpt[String].fold[JsResult[WsClientMessage]](JsError(JsPath, "Type 'T' was not specified")) {
+      case WsPingOrPong.tpe         => json.validate[WsPingOrPong]
+      case WsOrderBookSubscribe.tpe => json.validate[WsOrderBookSubscribe]
+      case WsAddressSubscribe.tpe   => json.validate[WsAddressSubscribe]
+      case WsUnsubscribe.tpe        => json.validate[WsUnsubscribe]
+      case x                        => JsError(JsPath \ "T", s"An unknown type: $x")
+    }
+  }
+
+  val wsClientMessageWrites: Writes[WsClientMessage] = Writes {
+    case x: WsPingOrPong         => Json.toJson(x)
+    case x: WsOrderBookSubscribe => Json.toJson(x)
+    case x: WsAddressSubscribe   => Json.toJson(x)
+    case x: WsUnsubscribe        => Json.toJson(x)
+    case x                       => throw new NotImplementedError(x.getClass.getName)
+  }
```

Comment:
Only using play-json discriminator. There is a way to specify a field as a discriminator, but I haven't found yet a way to specify the type as "pp" instead of "WsPingOrPong"

Reply:
I heard that circe can handle discriminators out of the box
Lets move to circe in brave soon future